Task 4-6. Barantsev

TiuAnna opened the pull request:
Задание №4: Сделать тесты для создания групп и контактов в адресной книге
Перед выполнением этого задания необходимо подготовить тестовый стенд и установить учебное приложение.
Повторить действия, которые были продемонстрированы в лекциях.
Затем сделать тест для создания нового контакта, проделав аналогичную последовательностью действий.

Задание №5: Реализовать двухуровневую архитектуру тестового набора
Повторить действия, продемонстрированные в лекциях, выполняя одновременно трансформацию тестов и для групп, и для контактов

Задание №6: Сделать тесты для удаления и модификации групп и контактов
Реализовать все недостающие (не реализованные в предыдущих домашних заданиях) тесты для создания, удаления и модификации групп и контактов, создавая попутно вспомогательные методы, какие потребуются

TiuAnna committed:
New class and methods are added

TiuAnna committed:
Bug is fixed

Missing method is added

TiuAnna committed:
Добавлен прогон теста по "1. Основы программирования"

Изначальный код изменен. Закомменченый код - первая попытка, неудачная, так как не нашла возможность имитировать в тесте ввод значения для сканера. Если есть решение - пишите

TiuAnna committed:
Test recorded with selenium is added

There is no changes in auto-created source code

TiuAnna committed:
Test for group creation is modified

Test class was renamed. Test actions were divided into methods.

TiuAnna committed:
Class TestBase is created

Common methods are moved to the class TastBase

TiuAnna committed:
Test for contact creation is added

Task N4 of Barantsev course

TiuAnna committed:
Deletion test is added, ApplicationManager class is created

All methods are moved to ApplicationManager class without dividing to smaller groups.
New packages were created: appmanager, tests and model

TiuAnna committed:
Helpers are created

ContactHelper, GroupHelper, SessionHelper, NavigationHelper were created. Related methods were moved there.

TiuAnna committed:
HelperBase class was added

New methods were extracted from the common actions and moved to HelperBase class

TiuAnna committed:
GroupModificationTests class was created

GroupModificationTests class was created using existing methods

TiuAnna committed:
ContactDeletionTests class was created

TiuAnna committed:
ContactModificationTests class was created

helga711 commented on addressbook-web-tests/src/test/java/pk/addressbook/appmanager/ContactHelper.java:
```diff
@@ -0,0 +1,42 @@
+        super(driver);
+    }
+    public void submitNewContactCreation() {
+        click(By.cssSelector("input:nth-child(87)"));
```
Можно проще By.name("submit"). Ниже еще есть такие локаторы, которые можно более наглядно прописать, на мой взгляд.

TiuAnna replied:
I hope you don't mind if I reply to you in English. I used the name that was defined by selenium, but if it passes with "submit" I will rather change to that one

helga711 commented on addressbook-web-tests/src/test/java/pk/addressbook/appmanager/ContactHelper.java:
```diff
@@ -0,0 +1,42 @@
+    public void deleteSelectedContacts() {
+        click(By.cssSelector(".left:nth-child(8) > input"));
+    }
+    public void acceptAlert() {
```
Мне кажется, лучше в HelperBase сразу вынести с текстом в параметрах, т.к. может еще где-то пригодиться.

TiuAnna committed:
input:nth-child(86) -> "submit"

Parameter value for submitContactModification() was changed to "submit"

TiuAnna committed:
Parameters values were changed

Parameters values for ContactHelper.deleteSelectedContacts(), .submitNewContactCreation() and .submitContactModification() were changed to By.name("delete"), By.name("submit") and By.name("update")

TiuAnna committed:
Merge branch 'test-for-wed-app-with-selenium' of https://github.com/TiuAnna/java_automation into test-for-wed-app-with-selenium
